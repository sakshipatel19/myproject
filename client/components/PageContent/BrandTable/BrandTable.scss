.brand-type-container {
	display: flex;
	flex-direction: column;
	margin-top: 50px;
	margin-bottom: 50px;
	box-sizing: border-box;
	border: 1px solid #e7e7e7;
	border-radius: 10px;
	background-color: #ffffff;
	box-shadow: 0 20px 60px -22px rgba(0, 67, 132, 0.3);
	.brand-type-heading-container {
		display: flex;
		justify-content: space-between;
		margin-top: 30px;
		margin-left: 3.4%;
		.download-icon {
			margin-right: 35px;
		}
		.label-container {
			color: #212129;
			font-family: "NotoSans-Bold";
			font-size: 22px;
			height: 32px;
			width: 78px;
			letter-spacing: 0px;
			line-height: 32px;
		}
	}
	.brand-type-detailpage-container {
		margin-top: 0px;
		margin-right: 3.4%;
		margin-bottom: 30px;
		.detail-page {
			padding-left: 30px;
			.detailpage-container {
				display: flex;
				flex-direction: column;
			}
		}
	}
	.brand-type-detailpage-container-sales {
		.detail-page {
			padding-left: 0px;
		}
	}
	.detailpage-table-body-sales {
		.brand-type-row-header {
			display: flex;
			font-family: "NotoSans-Condensed";
			font-size: 12px;
			padding: 20px 10px;
			.brand-type-col-1,
			.brand-type-col-3,
			.brand-type-col-4 {
				flex: 1;
			}
			.brand-type-col-2 {
				flex: 3;
			}
			.brand-type-col-1 {
				padding-right: 10px;
				text-align: right;
			}
			.brand-type-col-3,
			.brand-type-col-4 {
				padding: 0 10px;
				text-align: left;
			}
			.sort-icon {
				margin-left: 3px;
				position: absolute;
				.icon-class {
					visibility: hidden;
					margin-left: 3px;
					margin-top: 3px;
				}
				.icon-class.visible {
					visibility: visible;
				}
			}
			.sortSelected {
				color: #16abe0;
			}
		}
		.brand-type-row-content {
			max-height: 335px;
			overflow-y: scroll;
			font-family: "NotoSans-Regular";
			font-size: 14px;
			.brand-type-row {
				display: flex;
				align-items: center;
				padding: 8px 10px;
				.brand-type-col-1,
				.brand-type-col-2,
				.brand-type-col-4 {
					flex: 1;
				}
				.brand-type-col-3 {
					flex: 3;
				}
			}
			.brand-type-col-1 {
				text-align: right;
				padding-right: 10px;
				white-space: normal;
			}
			.brand-type-col-4,
			.brand-type-col-2 {
				padding: 0 10px;
				text-align: left;
				.dot-container {
					&.normal {
						margin-right: 10px;
						min-width: 35%;
						display: inline-block;
					}
				}
				.dot-text {
					font-size: 14px;
				}
			}

			// .brand-type-col-2.compare-selected {
			// 	width: 22%;
			// 	display: flex;
			// 	margin-right: 10px;
			// 	.dots-container {
			// 		width: 100%;
			// 		display: flex;
			// 		flex-direction: row;
			// 		justify-content: space-between;
			// 		.dot-container {
			// 			width: 50%;
			// 			margin-left: 10px;
			// 			.dot-text {
			// 				font-size: 12px;
			// 				text-align: right;
			// 			}
			// 		}
			// 	}
			// }
			// .brand-type-col-4 {
			// 	width: 265px;
			// 	display: flex;
			// 	justify-content: flex-end;
			// 	.dots-container .dot-container {
			// 		margin-left: 10px;
			// 	}
			// }
			// .brand-type-col-3.compare-linear-progressbar {
			// 	display: flex;
			// 	flex-direction: column;
			// 	width: 55.6%;
			// 	margin: 0 2.17% 0 2.17%;
			// 	.linear-progressbar {
			// 		margin-bottom: 4px;
			// 	}
			// }
			// .brand-type-col-4.compare-selected {
			// 	width: 30%;
			// 	display: flex;
			// 	text-align: right;
			// 	margin-right: 10px;
			// 	.dots-container {
			// 		.dot-container {
			// 			margin-left: 10px;
			// 			.dot-text {
			// 				font-size: 12px;
			// 			}
			// 		}
			// 	}
			// }
		}
	}
	.detailpage-table-body {
		width: 100%;
		align-content: center;
		justify-content: space-between;
		margin-top: 17px;
		.brand-type-row {
			display: flex;
			justify-content: space-between;
			align-items: center;
			color: #212129;
			font-family: "NotoSans-Regular";
			letter-spacing: 0px;
			font-size: 14px;
			line-height: 20px;
			padding: 8px 10px;
			.brand-type-col-1 {
				display: flex;
				justify-content: flex-end;
				width: 10.62%;
				text-align: right;
			}
			.sort-icon {
				margin-left: 3px;
				.icon-class {
					visibility: hidden;
					margin-left: 3px;
					margin-top: 3px;
				}
				.icon-class.visible {
					visibility: visible;
				}
			}
			.brand-type-col-2 {
				display: flex;
				width: 2.76%;
				margin-right: 11.62%;
				.dots-container {
					display: flex;
					width: 97px;
					margin: 0 0 8px -17px;
					.dot-container {
						display: flex;
						align-items: center;
						margin-left: 20px;
						.dot-text {
							padding-bottom: 1px;
						}
					}
				}
			}

			.brand-type-col-3 {
				display: flex;
				flex-direction: column;
				width: 66.6%;
				margin: 0 3.17% 0 3.17%;
				.linear-progressbar:last-child {
					margin-top: 4px;
				}
			}
			.brand-type-col-3.compare-linear-progressbar {
				margin-top: 4px;
			}
		}
		.brand-type-row-content {
			max-height: 251px;
			overflow-y: auto;
			.brand-type-col-2 {
				padding-left: 1%;
			}
			&::-webkit-scrollbar {
				width: 10px;
				background-color: #ffffff;
				border-radius: 4px;
			}
			&::-webkit-scrollbar-thumb {
				width: 8px;
				border-radius: 4px;
				background-color: #9dcafb;
			}
			&::-webkit-scrollbar-track {
				background-color: #f7f9fc;
			}
		}
		.brand-type-row.header {
			padding-bottom: 11px;
			margin-bottom: 27px;
			font-family: "NotoSans-Condensed";
			.brand-type-col-1,
			.brand-type-col-2,
			.brand-type-col-4 {
				font-size: 12px;
				line-height: 18px;
			}
			.brand-type-col-1 {
				width: 12%;
				padding-right: 0.6%;
			}
			.brand-type-col-1:hover,
			.brand-type-col-2:hover {
				color: #16abe0;
				cursor: pointer;
			}
			.sortSelected {
				color: #16abe0;
				display: flex;
			}
			.disableSort {
				pointer-events: none;
			}
			border-bottom: 2px solid #f2f2f2;
		}
	}
}

.brandtable-container {
	margin-bottom: 50px;
	.card-container {
		padding: 30px 3.44% 14px 3.44%;
		.brandtable-detailpage-container {
			margin-top: 30px;
			max-height: 335px;
			overflow-y: auto;
			&::-webkit-scrollbar {
				width: 10px;
				background-color: #ffffff;
				border-radius: 4px;
			}
			&::-webkit-scrollbar-thumb {
				width: 8px;
				border-radius: 4px;
				background-color: #9dcafb;
			}
			&::-webkit-scrollbar-button {
				height: 40px;
			}
			&::-webkit-scrollbar-button:increment {
				height: 0px;
			}
			&::-webkit-scrollbar-track {
				background-color: #f7f9fc;
			}
			.common-table-container {
				border-bottom: 2px solid transparent;
				th {
					position: sticky;
					top: 0;
					background-color: white;
					box-shadow: 0 2px #f2f2f2;
				}
				td {
					border-bottom: 2px solid transparent;
				}
				tbody {
					width: 100%;
					.table-cell-brandName {
						text-align: left;
					}
					.table-cell-averageDiscount {
						text-align: right;
						padding-right: 5%;
					}
					tr:hover {
						background-color: transparent;
					}
					.compare-linear-progressbar {
						margin-top: -6px;
					}
					.progressbar-container-with-dots {
						display: flex;
						.progressbar-container {
							display: flex;
							flex-direction: column;
							margin: 11px 6.17% 0 0;
						}
					}
					.dots-container.compare {
						display: flex;
						.dot-container {
							min-width: 37px;
							display: flex;
							align-items: center;
							.dot-text {
								margin-left: 4px;
							}
						}
						.dot-container:first-child {
							margin-right: 15%;
						}
					}
				}
			}
		}
	}
	.brandtable-loading-indicator {
		max-height: 335px;
	}
}
