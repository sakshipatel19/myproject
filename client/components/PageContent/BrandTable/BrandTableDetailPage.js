import React, { Component, Fragment } from "react";
import "./BrandTable.scss";
import LinearProgressBar from "../../common/LinearProgressBar";
import DotWithLabel from "../../common/DotWithLabel";
import Icon from "../../common/Icon";
import { applySort } from "../../../utils/array";

const SORT_ORDER = {
	ascendingOrder: "ASC",
	descendingOrder: "DESC"
};

class BrandTableDetailPage extends Component {
	state = {
		showBestSellerBrandDetailPage: true,
		sortType: "ASC",
		sortBy: "label"
	};

	createTableHeaderObject = () => {
		return [
			{
				name: "Name",
				key: "label"
			},
			{
				name: "Score",
				key: "score",
				isNumber: true
			}
		];
	};

	bindHeaderRowData = (arrObj, dataLength) => {
		const headersList = arrObj.map((e, i) => {
			return (
				<div
					className={`${`brand-type-col-${i + 1}`} ${this.state.sortBy === e.key && dataLength > 5 ? "sortSelected" : ""
						} ${e.isNumber ? "number-text" : ""} ${dataLength <= 5 ? "disableSort" : ""
						}`}
					onClick={() => this.onSortChange(e.key)}
					key={`${e.key}`}
				>
					{e.name}
					<span className="sort-icon">
						<Icon
							iconClass={`${this.state.sortBy === e.key &&
								dataLength > 5 &&
								"visible"}`}
							name={this.state.sortType === "ASC" ? "sortUp" : "sortDown"}
							size={8}
						/>
					</span>
				</div>
			);
		});

		return <div className="brand-type-row header">{headersList}</div>;
	};

	onSortChange = sortBy => {
		let currentsortOrder = this.state.sortType;
		let sortType = SORT_ORDER.ascendingOrder;
		let currentSort = this.state.sortBy;
		if (currentSort === sortBy)
			sortType =
				currentsortOrder === SORT_ORDER.ascendingOrder
					? SORT_ORDER.descendingOrder
					: SORT_ORDER.ascendingOrder;
		this.setState({ sortType, sortBy });
	};

	createBrandTableBody = data => {
		const brandTableList = data?.brandScoreInitial || [];
		const brandList = brandTableList.map((item, i) => {
			return (
				<div className="brand-type-row" key={`${item?.label}_${i}`}>
					<div className={`brand-type-col-1`}>{item.label}</div>
					{item.compareScore !== null ? (
						<div className="brand-type-col-3 compare-linear-progressbar">
							<LinearProgressBar
								rounded
								animate
								percent={item.score / 100}
								width={674}
								height={5}
								backgroundColor="#f3f5f8"
								fillColor={"#4CA0FA"}
							/>
							<LinearProgressBar
								rounded
								animate
								percent={item.compareScore / 100}
								width={674}
								height={4.8}
								backgroundColor="#f3f5f8"
								fillColor="#F79764"
							/>
						</div>
					) : (
							<div className="brand-type-col-3">
								<LinearProgressBar
									rounded
									animate
									percent={item.score / 100}
									width={674}
									height={5}
									backgroundColor="#f3f5f8"
									fillColor={"#4CA0FA"}
								/>
							</div>
						)}
					<div className={`brand-type-col-2`}>
						{this.props.compareSelected ? (
							<div className="dots-container">
								<Fragment>
									<DotWithLabel
										color="#4CA0FA"
										text={item?.score === null ? "NA" : `${item?.score}%`}
									/>
									<DotWithLabel
										color="#F79764"
										text={
											item?.compareScore === null
												? "NA"
												: `${item?.compareScore}%`
										}
									/>
								</Fragment>
							</div>
						) : (
								<div>
									{item?.score === null || item?.score === undefined
										? "NA"
										: `${item?.score}%`}
								</div>
							)}
					</div>
				</div>
			);
		});
		return <div className="brand-type-row-content">{brandList}</div>;
	};

	render() {
		let data = this.props.data || [];
		let brandScoreInitial = data?.brandScoreInitial || [];
		applySort(
			brandScoreInitial,
			this.state.sortBy,
			this.state.sortType,
			SORT_ORDER
		);
		return (
			<div className="detailpage-container">
				{this.state.showBestSellerBrandDetailPage && (
					<div className="detailpage-table-body">
						{this.bindHeaderRowData(
							this.createTableHeaderObject(),
							brandScoreInitial?.length
						)}
						{this.createBrandTableBody(data)}
					</div>
				)}
			</div>
		);
	}
}

export default BrandTableDetailPage;
