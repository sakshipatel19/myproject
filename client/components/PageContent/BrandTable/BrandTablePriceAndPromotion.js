import React, { Component, Fragment } from "react";
import "./BrandTable.scss";
import Card from "../../common/Card";
import Table from "../../common/Table";
import LoadingIndicator from "../../common/LoadingSkeleton/LoadingSquare";
import DataLoadError from "../common/DataLoadError/DataLoadError";
import LinearProgressBar from "../../common/LinearProgressBar";
import DotWithLabel from "../../common/DotWithLabel";

const tableFilter = {
	sortBy: "brandName",
	sortType: "ASC",
	pageNo: 1,
	pageSize: 15
};

class BrandTable extends Component {
	getProgressbarComponent = (initialScore, compareScore) => {
		return (
			<div className="progressbar-container-with-dots">
				<div className="progressbar-container">
					<LinearProgressBar
						rounded
						animate
						percent={initialScore / 100}
						width={195}
						height={5}
						backgroundColor="#f3f5f8"
						fillColor="#4CA0FA"
					/>
					{compareScore !== null && (
						<div className="compare-linear-progressbar">
							<LinearProgressBar
								rounded
								animate
								percent={compareScore / 100}
								width={195}
								height={5}
								backgroundColor="#f3f5f8"
								fillColor="#F79764"
							/>
						</div>
					)}
				</div>
				{this.getLabelWithDotsContainer(initialScore, compareScore)}
			</div>
		);
	};
	getLabelWithDotsContainer = (initialScore, compareScore) => {
		return (
			<div
				className={`dots-container ${this.props.compareSelected ? "compare" : ""
					}`}
			>
				{this.props.compareSelected ? (
					<Fragment>
						<DotWithLabel
							size={5}
							color="#4CA0FA"
							text={initialScore === null ? "NA" : `${initialScore}%`}
						/>
						<DotWithLabel
							size={5}
							color="#F79764"
							text={compareScore === null ? "NA" : `${compareScore}%`}
						/>
					</Fragment>
				) : (
						<div className="dot-text">
							{initialScore === null ? "NA" : `${initialScore}%`}
						</div>
					)}
			</div>
		);
	};
	DataRowComponent = row => {
		const { tableHeader } = this.props;

		return (
			<tr className="aligncenter">
				{tableHeader.map((col, i) => {
					switch (col.key) {
						case "brandName":
							return (<td className={"table-cell-brandName"} key={i}>{row.brandName}</td>)
							break;
						case "averageDiscount":
							return (<td className={"table-cell-averageDiscount"} key={i}>
								{this.getLabelWithDotsContainer(
									row.averageDiscount.initial,
									row.averageDiscount.compare
								)}
							</td>)
							break;
						default:
							return (<td className={"table-cell"} key={i}>
								{this.getProgressbarComponent(
									row[col.key].initial,
									row[col.key].compare
								)}
							</td>)
							break;

					}

				})
				}
			</tr>
		);
	};

	createUniqueKeyString = obj => `${obj.brandName}`;

	disableSortingInHeader = (headers) => {
		headers.map(header => {
			header.isSortable = false;
		});
		return headers;
	};
	render() {
		const { data, onRetryClick, enable, tableHeader } = this.props;
		const brandsPriceAndPromotionScoreList =
			data?.data?.brandsPriceAndPromotionScoreList || [];
		const header =
			brandsPriceAndPromotionScoreList?.length <= 5
				? this.disableSortingInHeader(tableHeader)
				: tableHeader;
		return (enable ?
			<div className="brandtable-container section-header-main">
				<Card>
					<div className={"card-header"}>
						<div className="card-title section-header-title">{"Brands"}</div>
					</div>
					{data.fetching ? (
						<LoadingIndicator className="brandtable-loading-indicator" />
					) : data.error ? (
						<DataLoadError handleRetry={onRetryClick} />
					) : (
								<div className="brandtable-detailpage-container">
									<Table
										header={header}
										data={brandsPriceAndPromotionScoreList}
										tableFilter={tableFilter}
										DataRowComponent={this.DataRowComponent}
										createUniqueKeyString={this.createUniqueKeyString}
										isInternalSorting
									/>
								</div>
							)}
				</Card>
			</div> : null
		);
	}
}

export default BrandTable;
