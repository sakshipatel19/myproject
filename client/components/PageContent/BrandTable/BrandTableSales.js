import React, { Component, Fragment } from "react";
import "./BrandTable.scss";
import LinearProgressBar from "../../common/LinearProgressBar";
import DotWithLabel from "../../common/DotWithLabel";
import Icon from "../../common/Icon";
import { applySort } from "../../../utils/array";
import Card from "../../common/Card";
import { isNullOrUndefined, numberFormat, styles } from "../../../utils/number";
import LoadingTable from "../../common/LoadingSkeleton/LoadingTable";
import DataLoadError from "../common/DataLoadError/DataLoadError";

const SORT_ORDER = {
	ascendingOrder: "ASC",
	descendingOrder: "DESC"
};

class BrandTableDetailPage extends Component {
	state = {
		showBestSellerBrandDetailPage: true,
		sortType: "ASC",
		sortBy: "label"
	};

	createTableHeaderObject = () => {
		return [
			{
				name: "Name",
				key: "brandName",
				sortable: true
			},
			{
				name: "",
				key: "",
				sortable: false
			},
			{
				name: "Rank",
				key: "salesScore",
				isNumber: true,
				sortable: true
			},
			{
				name: "Category Size",
				key: "categorySize",
				isNumber: true,
				sortable: true
			}
		];
	};

	bindHeaderRowData = (arrObj, dataLength) => {
		const headersList = arrObj.map((e, i) => {
			return (
				<div
					className={`${`brand-type-col-${e.key == "categorySize" ? 4 : i + 1
						}`} ${this.state.sortBy === e.key && dataLength > 5 ? "sortSelected" : ""
						} ${e.isNumber ? "number-text" : ""} ${dataLength <= 5 ? "disableSort" : ""
						} ${this.props.compareSelected ? "compare" : ""}`}
					onClick={() => this.onSortChange(e.key)}
					key={`${e.key}`}
				>
					{e.name}
					{e.sortable && <span className="sort-icon">
						<Icon
							iconClass={`${this.state.sortBy === e.key &&
								dataLength > 5 &&
								"visible"}`}
							name={this.state.sortType === "ASC" ? "sortUp" : "sortDown"}
							size={8}
						/>
					</span>}
				</div>
			);
		});

		return <div className="brand-type-row-header">{headersList}</div>;
	};

	onSortChange = sortBy => {
		let currentsortOrder = this.state.sortType;
		let sortType = SORT_ORDER.ascendingOrder;
		let currentSort = this.state.sortBy;
		if (currentSort === sortBy)
			sortType =
				currentsortOrder === SORT_ORDER.ascendingOrder
					? SORT_ORDER.descendingOrder
					: SORT_ORDER.ascendingOrder;
		this.setState({ sortType, sortBy });
	};

	createBrandTableBody = data => {
		const brandTableList = data?.brandsSalesScoreAndRevenueList || [];
		const { isBrandCompare, compareBrands } = this.props;
		const progressBarWidth = this.viewportWidth <= 1375 ? 400 : 500;
		const brandList = brandTableList.map((item, i) => {
			const { brandName, salesScore, categorySize } = item;
			return (
				<div className="brand-type-row" key={`${brandName}_${i}`}>
					<div className={`brand-type-col-1`}>{brandName}</div>

					{isBrandCompare && item?.salesScore?.compare !== null ? (
						<div
							className={`brand-type-col-3 ${this.props.compareSelected ? "compare-linear-progressbar" : ""
								}`}
						>
							<LinearProgressBar
								rounded
								animate
								percent={item?.salesScore?.compare / 100}
								width={progressBarWidth}
								height={5}
								backgroundColor="#f3f5f8"
								fillColor={isBrandCompare ? compareBrands[i]?.color : "#4CA0FA"}
							/>
						</div>
					) : salesScore?.compare !== null ? (
						<div className="brand-type-col-3 compare-linear-progressbar">
							<LinearProgressBar
								rounded
								animate
								percent={salesScore?.initial / 100}
								width={progressBarWidth}
								height={5}
								backgroundColor="#f3f5f8"
								fillColor="#4CA0FA"
							/>

							<LinearProgressBar
								rounded
								animate
								percent={salesScore?.compare / 100}
								width={progressBarWidth}
								height={5}
								backgroundColor="#f3f5f8"
								fillColor="#F79764"
							/>
						</div>
					) : (
								<div
									className={`brand-type-col-3 ${this.props.compareSelected ? "compare-linear-progressbar" : ""
										}`}
								>
									<LinearProgressBar
										rounded
										animate
										percent={salesScore?.initial / 100}
										width={progressBarWidth}
										height={5}
										backgroundColor="#f3f5f8"
										fillColor={isBrandCompare ? compareBrands[i]?.color : "#4CA0FA"}
									/>
								</div>
							)}
					<div
						className={`brand-type-col-2 ${this.props.compareSelected ? "compare-selected" : ""
							}`}
					>
						{this.props.compareSelected ? (
							<div className="dots-container">
								<Fragment>
									<DotWithLabel
										color="#4CA0FA"
										text={
											isNullOrUndefined(salesScore?.initial)
												? "NA"
												: `${salesScore?.initial}%`
										}
										className="normal"
									/>
									<DotWithLabel
										color="#F79764"
										text={
											isNullOrUndefined(salesScore?.compare)
												? "NA"
												: `${salesScore?.compare}%`
										}
										className="compare"

									/>
								</Fragment>
							</div>
						) : (
								<div>
									{isNullOrUndefined(salesScore?.initial)
										? "NA"
										: `${salesScore?.initial}%`}
								</div>
							)}
					</div>
					<div
						className={`brand-type-col-4 ${this.props.compareSelected ? "compare-selected" : ""
							}`}
					>
						{this.props.compareSelected ? (
							<div className="dots-container">
								<Fragment>
									<DotWithLabel
										color="#4CA0FA"
										text={
											isNullOrUndefined(categorySize?.initial)
												? "NA"
												: this.categorySizeFormat(categorySize?.initial)
										}
										className="normal"
									/>
									<DotWithLabel
										color="#F79764"
										text={
											isNullOrUndefined(categorySize?.compare)
												? "NA"
												: this.categorySizeFormat(categorySize?.compare)
										}
										className="compare"
									/>
								</Fragment>
							</div>
						) : (
								<div>
									{isNullOrUndefined(categorySize?.initial)
										? "NA"
										: this.categorySizeFormat(categorySize?.initial)}
								</div>
							)}
					</div>
				</div>
			);
		});
		return <div className="brand-type-row-content">{brandList}</div>;
	};
	categorySizeFormat = score => {
		return numberFormat(score, this.props.countryCode, styles.decimal);
	};
	render() {
		const { brandsScore, enable } = this.props;
		let data = brandsScore.data || [];
		let brandScoreInitial = data?.brandsSalesScoreAndRevenueList || [];
		this.viewportWidth = window.innerWidth;
		applySort(
			brandScoreInitial,
			this.state.sortBy,
			this.state.sortType,
			SORT_ORDER
		);
		return (enable ?
			<Card className="brand-type-container">
				<div className={"card-header"}>
					<div className="card-title section-header-title">{"Brands"}</div>
				</div>
				<div className="brand-type-detailpage-container-sales">
					{brandsScore?.fetching ? (
						<LoadingTable />
					) : brandsScore?.error ? (
						<DataLoadError />
					) : (
								<div className="detail-page">
									<div className="detailpage-container">
										{this.state.showBestSellerBrandDetailPage && (
											<div className="detailpage-table-body-sales">
												{this.bindHeaderRowData(
													this.createTableHeaderObject(),
													brandScoreInitial?.length
												)}
												{this.createBrandTableBody(data)}
											</div>
										)}
									</div>
								</div>
							)}
				</div>
			</Card> : null
		);
	}
}

export default BrandTableDetailPage;
