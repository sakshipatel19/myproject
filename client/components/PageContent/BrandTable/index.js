import React from "react";
import BrandTableDetailPage from "./BrandTableDetailPage";
import Label from "../../common/Label";
import Icon from "../../common/Icon";
import LoadingTable from "../../common/LoadingSkeleton/LoadingTable";
import DataLoadError from "../common/DataLoadError/DataLoadError";

const BrandTable = props => {
	const { brandsScore, fetchBrands, compareSelected, enable } = props;
	const brandsData = brandsScore?.data;
	return (enable ?
		<div className="brand-type-container section-header-main">
			<div className="brand-type-heading-container">
				<Label text="Brands" labelClass="section-header-title" />
				<div className="download-icon">
					<Icon name="download" size={16} />
				</div>
			</div>
			<div className="brand-type-detailpage-container">
				<div className="detail-page">
					{brandsScore?.fetching ? (
						<LoadingTable />
					) : brandsScore?.error ? (
						<DataLoadError />
					) : (
								<BrandTableDetailPage
									data={brandsData}
									fetchBrands={fetchBrands}
									compareSelected={compareSelected}
								/>
							)}
				</div>
			</div>
		</div> : null
	);
};

export default BrandTable;
