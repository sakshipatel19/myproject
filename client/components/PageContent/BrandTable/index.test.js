import React from "react";
import { shallow } from "enzyme";

import BrandTable from ".";

const brandsScore = {
    fetching:null,
    error:null,
    data:null
};
let baseProps = {
    fetchBrands:jest.fn(),
    brandsScore:brandsScore,
    compareSelected:false
};

describe("BrandTable tests", () => {
    it("BrandTable should render correctly", () => {
        const wrapper = shallow(<BrandTable {...baseProps} />);
        //calls componentWillReceiveProps
        wrapper.setProps({brandsData : brandsScore});
        expect(wrapper.state('brandsData')).toEqual(brandsScore);
        expect(wrapper).toMatchSnapshot();
    });
    it("BrandTable should render correctly when no props are received", () => {
        const wrapper = shallow(<BrandTable/>);
        expect(wrapper.state('brandsData')).toEqual([]);
    });
});
