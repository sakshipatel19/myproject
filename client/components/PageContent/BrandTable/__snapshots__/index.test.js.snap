// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandTable tests BrandTable should render correctly 1`] = `
<div
  className="brand-type-container"
>
  <div
    className="brand-type-heading-container"
  >
    <Label
      text="Brands"
    />
    <div
      className="download-icon"
    >
      <Icon
        name="download"
        size={16}
      />
    </div>
  </div>
  <div
    className="brand-type-detailpage-container"
  >
    <div
      className="detail-page"
    >
      <BrandTableDetailPage
        compareSelected={false}
        data={null}
        fetchBrands={[MockFunction]}
      />
    </div>
  </div>
</div>
`;
